lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/AccessControlRulesProvider.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import java.util.function.Supplier;
+
+public interface AccessControlRulesProvider<R>
```

Comment:
I think you could remove this class and instead bind to guice `Provider`. WDYT?

Reply:
Unfortunately, guice does not allow to bind to `Provider`. However, I can simply use `Supplier<R>`. I will get rid of this class by binding to `Supplier<R>`